components/script/dom/xr.rs

```diff
@@ -53,43 +56,79 @@ impl Drop for XR {
     }
 }
 
-impl XR {
+impl XRMethods for XR {
+    #[allow(unrooted_must_root)]
+    fn SupportsSessionMode(&self, mode: XRSessionMode) -> Rc<Promise> {
+        // XXXManishearth this should select an XR device first
+        let promise = Promise::new(&self.global());
+        if mode == XRSessionMode::Immersive_vr {
+            promise.resolve_native(&());
+        } else {
+            // XXXManishearth support other modes
+            promise.reject_error(Error::NotSupported);
+        }
+
+        promise
+    }
+
     #[allow(unrooted_must_root)]
-    pub fn get_displays(&self) -> Rc<Promise> {
+    fn RequestSession(&self, options: &XRSessionCreationOptions) -> Rc<Promise> {
         let promise = Promise::new(&self.global());
+        if options.mode != XRSessionMode::Immersive_vr {
+            promise.reject_error(Error::NotSupported);
+            return promise;
+        }
+
+        let displays = self.get_displays();
 
+        let displays = match displays {
+            Ok(d) => d,
+            Err(_) => {
+                promise.reject_native(&());
+                return promise;
+            },
+        };
+
+        if displays.is_empty() {
+            promise.reject_error(Error::Security);
+        }
+
+        let session = XRSession::new(&self.global(), &displays[0]);
+        promise.resolve_native(&session);
+
+        promise
+    }
+}
+
+impl XR {
+    #[allow(unrooted_must_root)]
```

Comment:
Why is this required?

Reply:
anything returning promises trips the lint

https://github.com/servo/servo/issues/22504

Reply:
Right, but this doesn't return a promise.

Reply:
oh, I was looking at the wrong impl block

This may be unnecessary, checking